### PR TITLE
PRIME-2750 HA site list enhancement 

### PR DESCRIPTION
**Pull request by bergomi02**

### CODE REVIEW

**Comment by neophyte57** [comment]
Running locally, I'm not seeing a Care Type I'm expecting, i.e. Custom Care 
image

image

**Comment by bergomi02** [comment]
> Running locally, I'm not seeing a Care Type I'm expecting, i.e. Custom Care image
> 
> image

@neophyte57 Can you please check if the care type is in CareTypeLookup. The care type in the dropdown is based on the lookup in CareTypeLookup table.

**Comment by neophyte57** [comment]
> > Running locally, I'm not seeing a Care Type I'm expecting, i.e. Custom Care image
> > image
> 
> @neophyte57 Can you please check if the care type is in CareTypeLookup? The care type in the dropdown is based on the lookup in CareTypeLookup table.

No it isn't:
image

Should it be based on `HealthAuthorityCareType` instead?
image

**Comment by bergomi02** [comment]
> > > Running locally, I'm not seeing a Care Type I'm expecting, i.e. Custom Care image
> > > image
> > 
> > 
> > @neophyte57 Can you please check if the care type is in CareTypeLookup? The care type in the dropdown is based on the lookup in CareTypeLookup table.
> 
> No it isn't: image
> 
> Should it be based on `HealthAuthorityCareType` instead? image

The Care Type is only for HA's site, the master list should contain all the Care Types from different HAs. It is how PRIME Admin can add new Care Type to HA. 
Beside, the code change here does not look at the records in HealthAuthorityCareType table. Instead, it takes CareTypeLookup as the base and filter out the ones not used in any of the HA sites.

**Comment by bergomi02** [comment]
@neophyte57 
The code is updated to pull the care type from the existing sites instead.

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts [hunk 1]
```diff
@@ -82,11 +105,62 @@ export class SiteManagementPageComponent implements OnInit {
...
+      .subscribe((sites: HealthAuthoritySiteList[]) => {
+        this.healthAuthoritySites = sites.sort((a, b) => a.siteName.toLocaleLowerCase().localeCompare(b.siteName.toLocaleLowerCase()))
+
+        const haVendors = this.healthAuthoritySites.map((s) => {
+          return s.healthAuthorityVendor.vendorCode;
+        });
+
+        const haCareTypes = this.healthAuthoritySites.map((s) => {
+          return s.healthAuthorityCareType.careType;
+        });
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts:122]: These should be sorted for the UI

Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts:122]: Yes, the code below it is to store the value in the UI variable with out duplicate. haCareTypes contains a list of care types but many are duplicated.

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.html [hunk 2]
```diff
@@ -25,29 +25,60 @@
...
                       [menu]="healthAuthoritySiteMenu"
                       [menuOutletContext]="{ healthAuthoritySite: healthAuthoritySite, healthAuthorityId: healthAuthorityId }"
                       [properties]="[
-                            { key: 'Site Name', value: healthAuthoritySite?.siteName },
-                            { key: 'Site ID', value: healthAuthoritySite?.pec },
-                            { key: 'Vendor', value: healthAuthoritySite?.healthAuthorityVendor?.vendorCode | configCode : 'vendors' },
+                            { key: 'Site Name', value: healthAuthoritySite?.siteName, _50: true },
+                            { key: 'Site ID', value: healthAuthoritySite?.pec, _50: true },
+                            { key: 'Care Type', value: healthAuthoritySite?.healthAuthorityCareType?.careType, _50: true },
+                            { key: 'Vendor', value: healthAuthoritySite?.healthAuthorityVendor?.vendorCode | configCode : 'vendors', _50: true },
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.html:81]: Can you add a comment re: what ` _50: true` is for or link to documentation?

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts (hunk 1 above)
Comment (neophyte57) [reply prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts:122]: I mean drop-down list should be sorted
image

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts [hunk 3]
```diff
@@ -27,20 +31,39 @@ export class SiteManagementPageComponent implements OnInit {
   public busy: Subscription;
   public title: string;
   public healthAuthorityId: number;
-  public healthAuthoritySites$: Observable<HealthAuthoritySiteList[] | null>;
+  public healthAuthoritySites: HealthAuthoritySiteList[];
+  public displaySites: HealthAuthoritySiteList[];
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts:35]: Is `displaySites` used?

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts [hunk 4]
```diff
@@ -82,11 +105,62 @@ export class SiteManagementPageComponent implements OnInit {
...
 
   public ngOnInit(): void {
+    this.createFormInstance();
+    this.initForm();
     const authorizedUser = this.authorizedUserService.authorizedUser;
     this.healthAuthorityId = authorizedUser.healthAuthorityCode;
-    this.healthAuthoritySites$ = this.authorizedUserResource.getAuthorizedUserSites(authorizedUser.id);
+    this.authorizedUserResource.getAuthorizedUserSites(authorizedUser.id)
+      .subscribe((sites: HealthAuthoritySiteList[]) => {
+        this.healthAuthoritySites = sites.sort((a, b) => a.siteName.toLocaleLowerCase().localeCompare(b.siteName.toLocaleLowerCase()))
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts:114]: A site in progress may not have a Site Name yet

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts [hunk 5]
```diff
@@ -82,11 +105,62 @@ export class SiteManagementPageComponent implements OnInit {
...
+    this.careTypeCode.valueChanges.subscribe(() => {
+      this.filterSites();
+    })
+  }
+
+  public filterSites() {
+    const authorizedUser = this.authorizedUserService.authorizedUser;
+    this.authorizedUserResource.getAuthorizedUserSites(authorizedUser.id)
+      .subscribe((sites: HealthAuthoritySiteList[]) => {
+        this.healthAuthoritySites = sites.sort((a, b) => a.siteName.toLocaleLowerCase().localeCompare(b.siteName.toLocaleLowerCase()));
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts:149]: A site in progress may not have a Site Name yet

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts (hunk 1 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts:121]: My bad, misunderstand. It is fixed.

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts (hunk 3 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts]: Variable removed.

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.html (hunk 2 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.html:81]: Added comment.

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts (hunk 4 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts]: Good point. Fixed.

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts (hunk 5 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts]: Fixed.

**Review by neophyte57 — APPROVED** [review]
FYI, 11 HA Sites in PROD without Site ID ... none Approved yet  😊🤔
image

**Pull request merged**